woocommerce/woocommerce-android
# Add custom amounts business logic

Closes: #10016 


#### NOTE: Please follow the below PR orders for reviewing
This one!
https://github.com/woocommerce/woocommerce-android/pull/10054
https://github.com/woocommerce/woocommerce-android/pull/10057
https://github.com/woocommerce/woocommerce-android/pull/10062

### Description


This PR introduces the core business logic and UI components required for the "Custom Amounts" feature.

**Key Changes:**

**Business Logic for Custom Amounts:** Implemented the essential logic to handle the addition of custom amounts.

**Custom Amounts Adapter:** Introduced a new adapter to handle the display of added custom amounts within a RecyclerView.

**Custom Amount Item UI:** Designed a user-friendly and intuitive UI for each custom amount item.

### Testing instructions

1. Navigate to the `orders` tab
2. Click on "+" button to navigate the the order creation screen
3. Click on the "Add custom amount" button
4. Add a custom amount
5. Ensure the custom amount gets added
6. Ensure you see the "custom amount" subsection instead of fees
7. Ensure the order total is correct
8. Add any product and ensure the custom amount, and order total are correct.

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/1331230/cd57c9cc-f7e6-46c1-a95d-30ab5f1e4caf
